fix AutoDisconnectHandler from triggering on tab changes

yannickberk opened the pull request:
This fixes an issue where the SegmentedControl stops working when switching from a tab with a SegmentedControl and later navigating to the tab again.

The AutoDisconnectHandler was triggering when leaving a tab, because only the tabbedPages current element was returned by the GetNavigationTree method. This is problematic as of the time of evaluation the tabbedPage.CurrentElement is already set to the new tab that is navigated to.

GetNavigationTree now returns all contained tabs, to not prematurely disconnect the handlers.

yannickberk committed:
fix AutoDisconnectHandler from triggering on tab changes

This fixes an issue where the SegmentedControl stops working when switching from a tab with a SegmentedControl and later navigating to the tab again.

The AutoDisconnectHandler was triggering when leaving a tab, because only the tabbedPages current element was returned by the GetNavigationTree method. This is problematic as of the time of evaluation the tabbedPage.CurrentElement is already set to the new tab that is navigated to.

GetNavigationTree now returns all contained tabs, to not prematurely disconnect the handlers.

thomasgalliker commented:
@yannickberk thanks for the attention in this case. There is also another issue regarding the auto disconnect feature in branch #33. I first gonna merge #33 and then have a look at this one 👍🏻

yannickberk commented:
@thomasgalliker great thank you, i added a branch [reproducible-sample-for-tab-issue](https://github.com/yannickberk/Plugin.SegmentedControl.Maui/tree/reproducible-sample-for-tab-issue) to my fork, where i changed the sample app to use tabbed pages.
There you can reproduce and confirm the issue with changing tabs back and forth triggering the disconnect handlers and making the SegmentedControl unusable till the page gets recreated.

thomasgalliker commented:
@yannickberk can you update your fork with the latest develop branch of this repository and check if this solves the problem? I merged some changes from my own branch into develop.

thomasgalliker commented:
I created another PR #35  and an issue #36.
What currently doesn't work (and it seems very difficult to achieve) is that ALL SegmentedControlHandlers are disconnected automatically when a TabbedPage is disappearing. It looks like a MAUI design flaw that only the currently selected tab page receives the `Disconnected` event.

Can you @yannickberk check the latest nuget pre-release (1.4.32-pre) and let me know if it solves the TabbedPage problem?
We can continue the discussion in #36.

yannickberk committed:
Merge branch 'thomasgalliker:develop' into patch-1

yannickberk commented:
Incorporated proposed changes in #36 . This PR is no longer needed.
